File: CS/CalculatedPropertiesSolution.Module/BusinessObjects/Order.cs
using DevExpress.Persistent.Base;
using DevExpress.Persistent.BaseImpl;
using DevExpress.Xpo;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CalculatedPropertiesSolution.Module.BusinessObjects {
    [DefaultClassOptions]
    public class Order : BaseObject {
        public Order(Session session) : base(session) { }
        private string fDescription;
        public string Description {
            get { return fDescription; }
            set { SetPropertyValue("Description", ref fDescription, value); }
        }
        private decimal fTotal;
        public decimal Total {
            get { return fTotal; }
            set {
                bool modified = SetPropertyValue("Total", ref fTotal, value);
                if (!IsLoading && !IsSaving && Product != null && modified) {
                    Product.UpdateOrdersTotal(true);
                    Product.UpdateMaximumOrder(true);
                }
            }
        }
        private Product fProduct;
        [Association("Product-Orders")]
        public Product Product {
            get { return fProduct; }
            set {
                Product oldProduct = fProduct;
                bool modified = SetPropertyValue("Product", ref fProduct, value);
                if (!IsLoading && !IsSaving && oldProduct != fProduct && modified) {
                    oldProduct = oldProduct ?? fProduct;
                    oldProduct.UpdateOrdersCount(true);
                    oldProduct.UpdateOrdersTotal(true);
                    oldProduct.UpdateMaximumOrder(true);
                }
            }
        }
    }
}

File: VB/CalculatedPropertiesSolution.Module/BusinessObjects/Order.vb
Imports DevExpress.Persistent.Base
Imports DevExpress.Persistent.BaseImpl
Imports DevExpress.Xpo
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks

Namespace CalculatedPropertiesSolution.Module.BusinessObjects
    <DefaultClassOptions> _
    Public Class Order
        Inherits BaseObject

        Public Sub New(ByVal session As Session)
            MyBase.New(session)
        End Sub
        Private fDescription As String
        Public Property Description() As String
            Get
                Return fDescription
            End Get
            Set(ByVal value As String)
                SetPropertyValue("Description", fDescription, value)
            End Set
        End Property
        Private fTotal As Decimal
        Public Property Total() As Decimal
            Get
                Return fTotal
            End Get
            Set(ByVal value As Decimal)
                Dim modified As Boolean = SetPropertyValue("Total", fTotal, value)
                If Not IsLoading AndAlso Not IsSaving AndAlso Product IsNot Nothing AndAlso modified Then
                    Product.UpdateOrdersTotal(True)
                    Product.UpdateMaximumOrder(True)
                End If
            End Set
        End Property
        Private fProduct As Product
        <Association("Product-Orders")> _
        Public Property Product() As Product
            Get
                Return fProduct
            End Get
            Set(ByVal value As Product)
                Dim oldProduct As Product = fProduct
                Dim modified As Boolean = SetPropertyValue("Product", fProduct, value)
                If Not IsLoading AndAlso Not IsSaving AndAlso oldProduct IsNot fProduct AndAlso modified Then
                    oldProduct = If(oldProduct, fProduct)
                    oldProduct.UpdateOrdersCount(True)
                    oldProduct.UpdateOrdersTotal(True)
                    oldProduct.UpdateMaximumOrder(True)
                End If
            End Set
        End Property
    End Class
End Namespace
